crates/chia-sdk-driver/src/merkle_tree.rs

```diff
@@ -0,0 +1,179 @@
+use std::collections::HashMap;
+
+use chia_protocol::Bytes32;
+use sha2::{Digest, Sha256};
+
+const HASH_TREE_PREFIX: &[u8] = &[2];
+const HASH_LEAF_PREFIX: &[u8] = &[1];
+
+#[derive(Debug, Clone)]
+pub struct MerkleTree {
+    pub root: Bytes32,
+    pub proofs: HashMap<Bytes32, (u32, Vec<Bytes32>)>,
+}
+
+use std::fmt::Debug;
+
+#[derive(Debug, Clone)]
+pub enum BinaryTree<T> {
+    Leaf(T),
+    Node(Box<BinaryTree<T>>, Box<BinaryTree<T>>),
+}
+
+impl MerkleTree {
+    pub fn new(leaves: &[Bytes32]) -> Self {
+        if leaves.is_empty() {
+            return Self {
+                root: Bytes32::default(),
+                proofs: HashMap::new(),
+            };
+        }
+
+        let (root, proofs) = MerkleTree::build_merkle_tree(leaves);
+        Self { root, proofs }
+    }
+
+    fn build_merkle_tree(leaves: &[Bytes32]) -> (Bytes32, HashMap<Bytes32, (u32, Vec<Bytes32>)>) {
+        let binary_tree = MerkleTree::list_to_binary_tree(leaves).unwrap();
```

Comment:
Can this ever fail? Maybe return an `Option` or `Result` instead if so, we don't want driver code to panic in most cases

Reply:
I made it return `T::default` if you provide 0 items in the leaves vector - no more errors!